src/main/java/org/greenbuttonalliance/gbaresourceserver/usage/controller/exception/DefaultExceptionHandler.java

```diff
@@ -0,0 +1,45 @@
+/*
+ * Copyright (c) 2022 Green Button Alliance, Inc.
+ *
+ *  Licensed under the Apache License, Version 2.0 (the "License");
+ *  you may not use this file except in compliance with the License.
+ *  You may obtain a copy of the License at
+ *
+ *       https://www.apache.org/licenses/LICENSE-2.0
+ *
+ *  Unless required by applicable law or agreed to in writing, software
+ *  distributed under the License is distributed on an "AS IS" BASIS,
+ *  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *  See the License for the specific language governing permissions and
+ *  limitations under the License.
+ */
+
+package org.greenbuttonalliance.gbaresourceserver.usage.controller.exception;
+
+import jakarta.servlet.http.HttpServletRequest;
+import lombok.RequiredArgsConstructor;
+import lombok.extern.slf4j.Slf4j;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.http.HttpStatus;
+import org.springframework.web.bind.annotation.ControllerAdvice;
+import org.springframework.web.bind.annotation.ExceptionHandler;
+import org.springframework.web.bind.annotation.ResponseBody;
+import org.springframework.web.bind.annotation.ResponseStatus;
+
+@ControllerAdvice
+@ResponseBody
+@Slf4j
+@RequiredArgsConstructor(onConstructor = @__(@Autowired))
+public class DefaultExceptionHandler {
+
+	@ExceptionHandler(EntityNotFoundByIdException.class)
+	@ResponseStatus(HttpStatus.NOT_FOUND)
+	public String handleEntityNotFoundByIdException(EntityNotFoundByIdException e, HttpServletRequest req) {
+		return logAndReturnExternal(e, req);
+	}
+
+	private String logAndReturnExternal(GbaControllerException e, HttpServletRequest req) {
+		log.error("{} at {}", e.getInternalErrorMessage(), req.getRequestURI());
+		return e.getExternalErrorMessage();
+	}
```

Comment:
I don't think this needs to be expanded on any more, but I wanted to make something small that said "Hey, you probably typed in the wrong UUID" with this file and the `EntityNotFoundByIdException` for the API team and any tests they might do.

Reply:
Nice touch and yes it should help the API test implementors.